src/Components/Web.JS/src/Platform/WebAssemblyStartOptions.ts

```diff
@@ -10,11 +12,23 @@ export interface WebAssemblyStartOptions {
    */
   loadBootResource(type: WebAssemblyBootResourceType, name: string, defaultUri: string, integrity: string) : string | Promise<Response> | null | undefined;
 
+  /**
+   * Overrides the built-in way of loading blazor.boot.json
+   * @param environment The built-in environment setting on start.
+   */
+  loadBlazorBootJson(environment?: string) : BootConfigResult;
```

Comment:
What do you think about possibly merging this into the existing `loadBootResource` callback? We could add an extra `environment` parameter to it.

The existing semantics for `loadBootResource` are pretty good about falling back on default behavior for `type` values that are not known to the callback - they just return null/undefined and we fall back on the default.

Reply:
I've integrated this with the original loadBootResource callback.